bot/core/paginator.py

```diff
@@ -0,0 +1,421 @@
+import asyncio
+import logging
+import typing as t
+from textwrap import dedent
+
+import discord
+from discord import Embed
+from discord.abc import Messageable
+from discord.ext import commands
+from discord.ext.commands import Context, Paginator
+
+logger = logging.getLogger(__name__)
+
+
+class PaginatorError(Exception):
```

Comment:
Naming this `PaginationError` makes more sense since it signifies some problem which occurs while paginating not an error of the paginator itself

Note: You'll also need to rename all references to this class
```suggestion
class PaginationError(Exception):
```